lib/vrl/stdlib/src/find_table_row.rs

```diff
@@ -0,0 +1,191 @@
+use std::collections::BTreeMap;
+use vrl::{prelude::*, Condition};
+#[derive(Clone, Copy, Debug)]
+pub struct FindTableRow;
+impl Function for FindTableRow {
+    fn identifier(&self) -> &'static str {
+        "find_table_row"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "table",
+                kind: kind::BYTES,
+                required: true,
+            },
+            Parameter {
+                keyword: "condition",
+                kind: kind::OBJECT,
+                required: true,
+            },
+        ]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[]
+    }
+
+    fn compile(&self, state: &state::Compiler, mut arguments: ArgumentList) -> Compiled {
+        let tables = state
+            .get_enrichment_tables()
+            .as_ref()
+            .map(|tables| {
+                tables
+                    .table_ids()
+                    .into_iter()
+                    .map(Value::from)
+                    .collect::<Vec<_>>()
+            })
+            .unwrap_or_else(Vec::new);
+
+        let table = arguments.required_enum("table", &tables)?.to_string();
+        let condition = arguments.required_object("condition")?;
+
+        Ok(Box::new(FindTableRowFn { table, condition }))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct FindTableRowFn {
+    table: String,
+    condition: BTreeMap<String, expression::Expr>,
+}
+
+impl Expression for FindTableRowFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let condition = self
+            .condition
+            .iter()
+            .map(|(key, value)| {
+                Ok(Condition::Equals {
+                    field: key.to_string(),
+                    value: value.resolve(ctx)?.try_bytes_utf8_lossy()?.to_string(),
+                })
```

Comment:
Can `Condition` take a `&str` (or `Bytes`) to avoid the cloning here?

Reply:
`field` can. `value` is a bit more complicated.